// graphql-static-binding/dist/generators/binding-js.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var graphcool_js_1 = require("./graphcool-js");
exports.generator = {
    Main: renderMainMethod,
    Header: renderHeader,
};
function renderHeader(schema) {
    return "const { Binding: BaseBinding } = require('graphql-binding')\nconst { GraphQLResolveInfo } = require('graphql')";
}
function renderMainMethod(queryType, mutationType, subscriptionType) {
    return "module.exports.Binding = class Binding extends BaseBinding {\n  \n  constructor({ schema, fragmentReplacements }) {\n    super({ schema, fragmentReplacements });\n\n    var self = this\n    this.query = {\n" + graphcool_js_1.renderMainMethodFields('query', queryType.getFields()) + "\n    }" + (mutationType ? "\n      \n    this.mutation = {\n" + graphcool_js_1.renderMainMethodFields('mutation', mutationType.getFields()) + "\n    }" : '') + (subscriptionType ? "\n      \n    this.subscription = {\n" + graphcool_js_1.renderMainSubscriptionMethodFields('mutation', subscriptionType.getFields()) + "\n    }" : '') + "\n  }\n  \n  delegate(operation, field, args, context, info) {\n    return super.delegate(operation, field, args, context, info)\n  }\n}";
}
